test/recipes/80-test_ssl.t

```diff
@@ -603,10 +603,28 @@ sub testssl {
 	}
     };
 
+    subtest 'SNI tests' => sub {
+
+	plan tests => 7;
+
+      SKIP: {
+	  skip "TLSv1.0 is not supported by this OpenSSL build", 7
+	      if $no_tls1;
+
+	  ok(run(test([@ssltest, "-bio_pair", "-tls1", "-sn_client", "foo"])));
+	  ok(run(test([@ssltest, "-bio_pair", "-tls1", "-sn_server1", "foo"])));
+	  ok(run(test([@ssltest, "-bio_pair", "-tls1", "-sn_client", "foo", "-sn_server1", "foo", "-sn_expect1"])));
+	  ok(run(test([@ssltest, "-bio_pair", "-tls1", "-sn_client", "foo", "-sn_server1", "bar", "-sn_expect1"])));
+	  ok(run(test([@ssltest, "-bio_pair", "-tls1", "-sn_client", "foo", "-sn_server1", "foo", "-sn_server2", "bar", "-sn_expect1"])));
+	  ok(run(test([@ssltest, "-bio_pair", "-tls1", "-sn_client", "bar", "-sn_server1", "foo", "-sn_server2", "bar", "-sn_expect2"])));
+	  ok(run(test([@ssltest, "-bio_pair", "-tls1", "-sn_client", "foobar", "-sn_server1", "foo", "-sn_server2", "bar", "-sn_expect1"])));
```

Comment:
Yes, the result is supposed to be a no-op. Mostly making sure nothing bad happens (like a crash).